internal/lockclient/simple_client.go

```diff
@@ -24,18 +30,60 @@ var _ Client = (*SimpleClient)(nil)
 
 // SimpleClient implements Client, the lockclient for Lockey.
 type SimpleClient struct {
+	Id     uint8
+	config *SimpleConfig
 }
 
 // Acquire makes a HTTP call to the lockserver and acquires the lock.
 // The errors invloved may be due the HTTP errors or the lockservice errors.
 func (sc *SimpleClient) Acquire(d lockservice.Descriptors) error {
-	panic("TODO")
+	baseUrl := sc.config.IPAddr + ":" + sc.config.PortAddr
+	endPoint := baseUrl + "/acquire"
+
+	var jsonStr = []byte(`{"FileID":` + string(d) + "}")
+	req, err := http.NewRequest("POST", endPoint, bytes.NewBuffer(jsonStr))
+	req.Header.set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
```

Comment:
Im not too sure but is this closing before its opening? A "defer" deferres the function call until the callee function returns. So you can `defer Close` right after you open it.

Reply:
Same for below.

Reply:
https://golang.org/pkg/net/http/

Isn't this the standard way of doing it?

Reply:
Sure I was just confused where the body was.